Add smartcontract files

gertrudekerubo opened the pull request:
smart contracts and migration solidity files

Shamininski committed:
Add .gitignore and repair broken links in CONTRIBUTING.md documentation

gertrudekerubo committed:
add smart contract files

Shamininski reviewed (changes requested):
Let's fix the error on migrations first and make sure that everything runs correctly first.

Shamininski committed:
Update truffle-config.js

Add solc version parameter in truffle.config.js and change localhost to an IP address
